src/lib/logical_query_plan/lqp_translator.hpp

```diff
@@ -68,9 +69,8 @@ class LQPTranslator {
       const std::vector<std::shared_ptr<AbstractExpression>>& lqp_expressions,
       const std::shared_ptr<AbstractLQPNode>& node) const;
 
-  // Cache operator subtrees by LQP node to avoid executing operators below a diamond shape multiple times
-  mutable std::unordered_map<std::shared_ptr<const AbstractLQPNode>, std::shared_ptr<AbstractOperator>>
-      _operator_by_lqp_node;
+  // Cache operator subtrees by LQP node hash to avoid executing identical operators redundantly
+  mutable ska::bytell_hash_map<size_t, std::shared_ptr<AbstractOperator>> _operator_by_lqp_node;
```

Comment:
That's a bug, hashes can collide and then the wrong subplan gets inserted. LQP still need to be the key, use a custom hasher to extract the hash of the LQP from the shared ptr